### DISCUSSION TITLE
Are ss.alter and ss.set methods equivalent in Andes?

### DISCUSSION THREAD

**Discussion by sghqsads** [CURENT/andes#472] [Q&A] [answered]
Hello Hantao, Jinning,

I hope you're both doing well.

I've been exploring ANDES for some work and I've come across a couple of points where I could use your guidance.

I'm currently trying to figure out if the ss.alter and ss.set methods function equivalently within ANDES. I have been trying to use the ss.set method (for example, ss.GENROU.set) to modify the initial data, which is loaded from an xlsx file, before carrying out the power flow calculations and initialization with ANDES. However, I'm not certain if it's been successful.

In addition, I wanted to ask about the procedure after altering the initial data from the xlsx file. Would it be necessary to call ss.setup() again for these changes to take effect?

I really appreciate any guidance you can provide on these questions. Your insights are very valuable to me.

Best regards,
Tracy

**Comment by jinningwang** [comment, marked as answer]
The functions "alter" and "set" are designed to modify different properties of the model in ANDES. Specifically, "alter" is used to manipulate the property "v" of a model, whereas "set" can be applied to a wider range of properties, including ``v`` for values, ``a`` for address, and ``e`` for equation value, although it is less common for users to change the "a" or "e" properties.

You can find the source code for these functions in the ANDES repository at the following links: [alter](https://github.com/CURENT/andes/blob/d20c689f8b98195c75327a315ffca0ead3353272/andes/core/model/model.py#L476) and [set](https://github.com/CURENT/andes/blob/d20c689f8b98195c75327a315ffca0ead3353272/andes/core/model/model.py#L520).

As ANDES is an open-source software, all of its code is available for review and revision as needed.

**Reply by sghqsads** [reply]
Thank you. Jinning.
